modules/_labs/pagination/react/lib/useIsMobile.tsx

```diff
@@ -0,0 +1,17 @@
+import {useEffect, useState} from 'react';
+
+const useIsMobile = (threshold: number) => {
```

Comment:
Should we remove this hook? I feel like this is a very odd hook given that it is detecting whether or not something is below a certain threshold. This can probably be accomplished by using something more generic like: 

```tsx
import {useWindowSize} from 'react-use';

// In Pagination.tsx, instead of:
// const isMobile = useIsMobile(500);
const isMobile = useWindowSize().width < 500;
```

or I'm thinking we can use the matchMedia API (compatibility seems good):

```tsx
import {useMedia} from 'react-use';

// In Pagination.tsx, instead of:
// const isMobile = useIsMobile(500);
const isMobile = useMedia('(max-width: 500px)');
```

I would like to check with the design team as well to see if 500 absolutely needs to be hardcoded, I think it makes this component less useful when it is.

Reply:
I agree 500 is an odd number. The Modal specs call out `768` as the threshold for "mobile" - basically tablet or phone.

I don't know if there is official guidelines on how we detect "mobile"

Reply:
I opted to make Pagination take an optional width property. It makes more sense for Pagination to respond to a width than some random 'isMobile' input. The application can always pass in this width based on window innerWidth. This makes testing easier and makes the component more flexible. @nikolasjchaconas what do you think?